sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ConsistencyReader.java

```diff
@@ -235,7 +231,9 @@ public ConsistencyReader(
 
             case Any:
                 if (targetConsistencyLevel.v == ConsistencyLevel.SESSION) {
-                    return this.readSessionAsync(entity, desiredReadMode);
+                    return BackoffRetryUtility.executeRetry(
```

Comment:
we should consider doing a quick perf test for session consistency to see the perf impact of having retry logic hooked up here?

Reply:
Will do that before merging

Reply:
**_Results - are in the same ballpark - if any regression it is minimal (around 1%)....._**

**Baseline**
=============

_[2000000] operations performed in [296] seconds._

-- Timers ----------------------------------------------------------------------
latency
             count = 2000000
     1-minute rate = 6770.75 calls/second
     5-minute rate = 4997.96 calls/second
              95% <= 1.98 milliseconds
              98% <= 2.75 milliseconds
              99% <= 4.39 milliseconds
            99.9% <= 11.47 milliseconds


**Candidate (with SessionTokenMismatchRetryPolicy)**
==================================================

_[2000000] operations performed in [301] seconds._

-- Timers ----------------------------------------------------------------------
latency
             count = 2000000
     1-minute rate = 6691.68 calls/second
     5-minute rate = 4945.42 calls/second
              95% <= 2.09 milliseconds
              98% <= 3.16 milliseconds
              99% <= 4.52 milliseconds
            99.9% <= 11.01 milliseconds